src/main/webapp/app/localci/build-queue/build-queue.component.ts

```diff
@@ -259,4 +264,23 @@ export class BuildQueueComponent implements OnInit, OnDestroy {
     refresh() {
         this.loadFinishedBuildJobs();
     }
+
+    /**
+     * Update the build jobs duration
+     */
+    updateBuildJobDuration() {
+        if (!this.runningBuildJobs) {
+            return;
+        }
+
+        for (const buildJob of this.runningBuildJobs) {
+            if (buildJob.jobTimingInfo && buildJob.jobTimingInfo?.buildStartDate) {
+                const start = dayjs(buildJob.jobTimingInfo?.buildStartDate);
+                const now = dayjs();
+                buildJob.jobTimingInfo.buildDuration = now.diff(start, 'seconds');
+            }
+        }
+        // This is necessary to update the view when the build job duration is updated
+        this.runningBuildJobs = JSON.parse(JSON.stringify(this.runningBuildJobs));
```

Comment:
I think the proper way would be to trigger the change detection.
Or at least only create a new array instead of parsing it: `this.runningBuildJobs = [...this.runningBuildJobs];`

Reply:
These unfortunately don't work.
- ref.detectChanges() doesn't work due to how ngx-datatable is implemented
- `this.runningBuildJobs = [...this.runningBuildJobs];` didn't work too (although they say that it should work in their docs)

Reply:
The spread operator `('...')` is a shallow copy method, which means it copies the elements of the array but not the nested objects themselves, they remain referenced to the original objects. 
The `JSON.parse(...)` is a deep copying method and creates entirely new and independent objects. 
I did not go into detail on `BuildJobs -Class`, but this could explain why only the parsing works correctly for your `runningBuildJobs` array

Reply:
Since we already use eslint could we not just use the deep copy method of that library?
Also I'm not completely sure but could setting it to an empty array first and then re-setting it work?